### ISSUE TITLE
Keyboard shortcuts don't work in flameshot 0.10.2 on Fedora KDE wayland, but they worked in version 0.10.1

### ISSUE THREAD

**Issue by pizzadude** [flameshot-org/flameshot#2077]
0.10.2

In Fedora 35 KDE Wayland, keyboard shortcuts don't work with flameshot 0.10.2 from updates-testing repos. Meaning, I can't press esc, I can't Ctrl+Z to undo, etc. Downgrading to 0.10.1 fixes the issue. 

**To Reproduce**
Be on Fedora KDE 35 wayland session with latest packages from updates-testing repos, try to take a screenshot with flameshot, notice keyboard shortcuts don't work.

**Expected behavior**
For keyboard shortcuts to work in KDE wayland, like they did with version 0.10.1

**System Information**
Operating System: Fedora Linux 35
KDE Plasma Version: 5.23.2
KDE Frameworks Version: 5.87.0
Qt Version: 5.15.2
Kernel Version: 5.14.18-300.fc35.x86_64 (64-bit)
Graphics Platform: Wayland
Processors: 4 × Intel® Core™ i5-7300U CPU @ 2.60GHz
Memory: 7.6 GiB of RAM
Graphics Processor: Mesa Intel® HD Graphics 620

**Comment by mdxe** [comment]
I have Flameshot 0.10.1 and it also stopped working after the latest Arch Linux update... "Unable to capture screen" error.

**Comment by borgmanJeremy** [comment]
Update to update 10.2. Please don't clutter this issue with a seperate problem.

**Comment by pizzadude** [comment]
@borgmanJeremy What? This issue is with 10.2.

**Comment by borgmanJeremy** [comment]
Sorry about that! I think I replied to the wrong issue using the mobile interface. Apologies 😊

**Comment by mdxe** [comment]
You didn't replied to the wrong issue, but since you thought I was cluttering, I deleted my comment... (sorry for cluttering again)

**Comment by borgmanJeremy** [comment]
No problem, sorry if it was overly harsh, I hadn't had my coffee yet, should have been more polite.

**Comment by mdxe** [comment]
No problem, sorry for creating this chaos... I thought that Github would leave a trace of my deleted comment...

**Comment by borgmanJeremy** [comment]
Ive confirmed the issue is present on the master branch as well.

**Comment by nobeh** [comment]
I can also confirm that I'm having the same issue on:

- GNOME 40.4.0 + Wayland
- Ubuntu 21.10

**Comment by jas0n098** [comment]
This is caused by commit https://github.com/flameshot-org/flameshot/commit/5d66405ed0e0fb871c174b8c1dfffb4a4d48034e forcing QT_QPA_PLATFORM to xcb. Reverting this change makes everything work again, at least on KDE Wayland 5.23.3.

**Comment by borgmanJeremy** [comment]
When you revert that commit everything works fine on kde wayland?

**Comment by jas0n098** [comment]
> When you revert that commit everything works fine on kde wayland?

Yes, keyboard shortcuts as well as copy-pasting into Wayland and Xwayland clients works just fine. I've been using 0.10.1 and 0.10.2 (with the commit reverted) daily and haven't noticed any other issues.

**Comment by xvitaly** [comment]
> When you revert that commit everything works fine on kde wayland?

Fixed this issue on Fedora 35 by partial reverting of 5d66405ed0e0fb871c174b8c1dfffb4a4d48034e commit:

```diff
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -49,8 +49,7 @@ void wayland_hacks()
 {
     // Workaround to https://github.com/ksnip/ksnip/issues/416
     DesktopInfo info;
-    if ((info.windowManager() == DesktopInfo::GNOME) ||
-        (info.windowManager() == DesktopInfo::KDE)) {
+    if (info.windowManager() == DesktopInfo::GNOME) {
         qputenv("QT_QPA_PLATFORM", "xcb");
     }
 }
```

**Comment by pizzadude** [comment]
I can confirm that partially reverting that commit fixes the issue.

**Comment by xvitaly** [comment]
Added PR with fix: #2131.
